src/ceph-daemon

```diff
@@ -306,16 +306,32 @@ def get_config_and_keyring():
     return (config, keyring)
 
 def get_config_and_both_keyrings():
-    if args.config_and_keyrings == '-':
-        try:
-            j = injected_stdin
-        except NameError:
-            j = sys.stdin.read()
+    if args.config_and_keyrings:
+        if args.config_and_keyrings == '-':
+            try:
+                j = injected_stdin
+            except NameError:
+                j = sys.stdin.read()
+        else:
+            with open(args.config_and_keyrings, 'r') as f:
+                j = f.read()
+        d = json.loads(j)
+        (d.get('config'), d.get('keyring'), d.get('crash_keyring'))
     else:
-        with open(args.config_and_keyrings, 'r') as f:
-            j = f.read()
-    d = json.loads(j)
-    (d.get('config'), d.get('keyring'), d.get('crash_keyring'))
+        if args.key:
+            keyring = '[%s]\n\tkey = %s\n' % (args.name, args.key)
+        elif args.keyring:
+            with open(args.keyring, 'r') as f:
+                keyring = f.read()
+        else:
+            raise RuntimeError('no keyring')
+        crash_keyring = None
+        if args.crash_keyring:
+            with open(args.crash_keyring, 'r') as f:
+                crash_keyring = f.read()
+        with open(args.config, 'r') as f:
+            config = f.read()
+        return (config, keyring, crash_keyring)
```

Comment:
having no function arguments makes it really hard to see what is going on here.


```python
def get_config_and_both_keyrings(config_and_keyrings, key, name, keyring, crash_keyring, config):
    if config_and_keyrings:
        if config_and_keyrings == '-':
            try:
                j = injected_stdin
            except NameError:
                j = sys.stdin.read()
        else:
            with open(config_and_keyrings, 'r') as f:
                j = f.read()
        d = json.loads(j)
        (d.get('config'), d.get('keyring'), d.get('crash_keyring'))
    else:
        if key:
            keyring = '[%s]\n\tkey = %s\n' % (name, key)
        elif keyring:
            with open(keyring, 'r') as f:
                keyring = f.read()
        else:
            raise RuntimeError('no keyring')
        crash_keyring = None
        if crash_keyring:
            with open(crash_keyring, 'r') as f:
                crash_keyring = f.read()
        with open(config, 'r') as f:
            config = f.read()
        return (config, keyring, crash_keyring)
```

this is convoluted already need imo be splitted into proper functions.

Reply:
what is a function argument? :)